### DISCUSSION TITLE
Add horizontal option for templated list editor of data list

### DISCUSSION THREAD

**Discussion by dawoe** [leekelleher/umbraco-contentment#333] [Idea]
Now all options are displayed vertical.

Would be nice if we could have the option to display them horizontal.
This could be a config setting on the editor.

And then in the editor view instead of applying the class btn-group-vertical it could be  btn-group-justified

**Comment by leekelleher** [comment]
Hi @dawoe, apologies for not replying sooner.

What I've been doing to get the horizontal layout with Data List's Templated List is to add CSS classes to the container.

In the configuration, the **HTML attributes** field, I add a key of `class` and a value of `umb-property-editor--limit-width flex flex-wrap justify-between`.

![Screenshot of Data List's Templated List's HTML attributes field](https://github.com/leekelleher/umbraco-contentment/assets/209066/8d237dc4-5795-48e7-80ac-8eeac2fd3fe3)

_(of course, not very visible in a screengrab!)_

To note, `umb-property-editor--limit-width` restricts the width of the property-editor (to 800px), the rest are Umbraco's own flexbox classes.

With all that said, I totally appreciate that this is a "sure, that's great if you know the CSS classes!" ... and a toggle would be much more friendly. 👍  I shall take a look when I'm next working on Data List.  _I just wanted to offer an immediate workaround if it was needed._

**Reply by leekelleher** [reply]
For anyone wanting to see the full configuration example...

Screenshot of Data List's Templated List configuration.

![Screenshot of Data List's Templated List configuration](https://github.com/leekelleher/umbraco-contentment/assets/209066/922b81f0-8057-4106-a506-266e5b9d45e0)

Screenshot of Data List using the Templated List (with the Number Range data-source).

![Screenshot of Data List using the Templated List](https://github.com/leekelleher/umbraco-contentment/assets/209066/0a97054f-01a7-416c-b429-eef7ab16a6e7)

**Comment by leekelleher** [comment]
Closing this off, as added an `orientation` field to Templated List in Contentment v6.
